src/runtime/builtin_modules/builtins.cpp

```diff
@@ -112,30 +112,38 @@ extern "C" Box* binFunc(Box* x) {
     return r;
 }
 
-extern "C" Box* hexFunc(Box* x) {
-    static BoxedString* hex_str = getStaticString("__hex__");
-    CallattrFlags callattr_flags{.cls_only = true, .null_on_nonexistent = true, .argspec = ArgPassSpec(0) };
-    Box* r = callattr(x, hex_str, callattr_flags, NULL, NULL, NULL, NULL, NULL);
-    if (!r)
-        raiseExcHelper(TypeError, "hex() argument can't be converted to hex");
-
-    if (!PyString_Check(r))
-        raiseExcHelper(TypeError, "__hex__() returned non-string (type %.200s)", r->cls->tp_name);
+extern "C" Box* hexFunc(Box* self) {
```

Comment:
this functions use the CAPI way to raise exceptions but this functions are registered as throwing c++ exceptions...
So you can either register them as capi functions (take a look at "next" in the same file, you have to pass CAPI to ```FunctionMetadata::create``` and please declare the function as ```noexcept``` too to make sure that no exception accidentally slips through)
or you have to convert them to use ``` raiseExcHelper```